### ISSUE TITLE
Server's lastModified is later than the actual (local) lastModified

### ISSUE THREAD

**Issue by ceskyDJ** [saber-notes/saber#182]
**Describe the bug**
Notes created on local device disappears (probably are removed) when I tap to "sync" button in the top right corner. I use custom NextCloud provider.

**To Reproduce**
1. Create new note
2. Close note and go to app's main screen
3. Tap "sync" button (in top right corner)
4. After sync the note disappears

**Expected behavior**
Note is uploaded to NextCloud and is still visible.

**Screenshots**
No screenshot needed.

**Device (probably doesn't matter, same behavior is on Debian PC running app via Flatpak):**
 - Device: Samsung Galaxy Tab 8
 - OS: Android 13
 - Version: 0.5.0

**Comment by ceskyDJ** [comment]
When using "official" NextCloud, files don't save at all. I only see them in app's "home" page. When I open them, they are empty.
![image](https://user-images.githubusercontent.com/48548230/209908053-89b93cc3-f5a3-4841-9017-971db491680b.png)

**Comment by ceskyDJ** [comment]
Hmm, interesting. It did only on linux machine. After app restart it works well.

**Comment by ceskyDJ** [comment]
I tried your NextCloud on both devices and it seems ok. Problem is probably with custom NextCloud provider.

**Comment by adil192** [comment]
I'm thinking this could be what happened:
1. You edited a note.
2. The file is saved (successfully).
3. The file is uploaded but failed for some reason, leaving an empty file on the server.
4. This empty file has a last-modified date newer than your local file. When you click the sync button, Saber compares the last-modified dates and downloads the empty file (deleting your local note).

This could be fixed by overriding the server's last-modified date to match the local file when uploaded. Then the server's file won't take priority and overwrite the local file.

**Comment by adil192** [comment]
Linking https://github.com/provokateurin/nextcloud-neon/issues/174

**Comment by ceskyDJ** [comment]
Oh, it works well in current version.

**Comment by ceskyDJ** [comment]
Oh, I'm sorry but it is not completely solved. I lost all my (testing, so it's ok right now) notes after update to v0.5.5 that caused a force update (at least I think, I really don't know what happened). Manual updates seemed to work on v0.5.4 but on v0.5.5 they 100% don't work at all.

**Comment by ceskyDJ** [comment]
Your ("official") NextCloud still works well. Interesting.

**Comment by adil192** [comment]
Please can you check if this is fixed in v0.9.3?

**Comment by ceskyDJ** [comment]
I did. Original problem described here was fixed but there is a catch. Syncing between devices doesn't work at all. I tried your NextCloud and mine, too.

How to reproduce (the same behavior is for whiteboard):

- (I don't know it is necessary) Log out and log in back on two devices (e.g. notebook with Debian and tablet with Android).
- Create a new note, name it and draw something on the first device (e.g. phone).
- Go to "All notes" page and tap/click on refresh button in the top right corner.
- Change devices (e.g. to notebook).
- Go to "All notes" page and tap/click on the same refresh button.
- Nothing happens, just animation of the button appears. New note isn't here.
